magda-esri-portal-connector/src/EsriPortal.ts

```diff
@@ -0,0 +1,563 @@
+import AsyncPage from "@magda/typescript-common/dist/AsyncPage";
+import EsriPortalUrlBuilder from "./EsriPortalUrlBuilder";
+import formatServiceError from "@magda/typescript-common/dist/formatServiceError";
+import { ConnectorSource } from "@magda/typescript-common/dist/JsonConnector";
+import retry from "@magda/typescript-common/dist/retry";
+import request from "@magda/typescript-common/dist/request";
+import {
+    ESRI_NSW_ORG,
+    ESRI_NSW_PORTAL
+} from "@magda/typescript-common/dist/session/SessionConsts";
```

Comment:
Can we pull this out into something a bit more generic that could be applied to more than just NSW?

If this still needs client-specific stuff hard-coded into it then it's not ready for the core Magda codebase yet, it should probably stay in terriajs/magda.

Reply:
I will try to make this configurable.

Reply:
done